kafka-connect/src/main/java/io/tabular/iceberg/connect/channel/Coordinator.java

```diff
@@ -147,7 +149,23 @@ private void doCommit(boolean partialCommit) {
         .stopOnFailure()
         .run(
             entry -> {
-              commitToTable(entry.getKey(), entry.getValue(), offsetsJson, vtts);
+              Pair<Table, Optional<String>> tableBranch = getTableAndBranch(entry.getKey());
+              if (tableBranch != null) {
+                Map<Integer, Long> lastCommittedOffsetsForTable =
+                    lastCommittedOffsetsForTable(
+                        tableBranch.first(), tableBranch.second().orElse(null));
+                for (int i = 0; i < entry.getValue().size(); i++) {
+                  List<Envelope> envelopeList = entry.getValue().get(i);
+                  if (i < entry.getValue().size() - 1) {
+                    commitToTable(
+                        entry.getKey(),
+                        envelopeList,
+                        resolveOffsetsJson(envelopeList, lastCommittedOffsetsForTable),
+                        vtts);
```

Comment:
I prefer to set this to `null` after reading the description.
> Finally there is a VTTS (valid-through timestamp) property indicating through what timestamp records have been fully processed, i.e. all records processed from then on will have a timestamp greater than the VTTS. This is calculated by taking the maximum timestamp of records processed from each topic partition, and taking the minimum of these. If any partitions were not processed as part of the commit then the VTTS is not set.

The last snapshot can still have the vtts since we have not touched that implementation.